kong/router/compat.lua

```diff
@@ -315,18 +316,24 @@ local function get_exp_and_priority(route)
 end
 
 
+local GROUPED_PATHS = {}
+
 -- group array-like table t by the function f, returning a table mapping from
 -- the result of invoking f on one of the elements to the actual elements.
 local function group_by(t, f)
-  local result = {}
-  for _, value in ipairs(t) do
+  local result = GROUPED_PATHS
```

Comment:
This would be only safe if the `result` is always used in `non-yielding` code / code that is blocked by mutexes. I have done some tests, and it may require with huge config that we introduce yielding. To prevent high max latencies.